sensu/plugin.go

```diff
@@ -289,20 +492,126 @@ func validateEvent(event *corev2.Event) error {
 
 // SetValue sets the configuration value based on either a raw string or
 // json-encoded string.
-func (p *PluginConfigOption[T]) SetValue(valueStr string) error {
-	switch value := (interface{}(p.Value)).(type) {
-	case *[]string:
-		if err := json.Unmarshal([]byte(valueStr), value); err == nil {
-			return nil
+func (p *PluginConfigOption[T]) SetValue(valueStr string) (err error) {
+	if p.Value == nil {
+		return errors.New("PluginConfigOption.Value not set!")
+	}
+	defer func() {
+		if err == nil {
+			err = p.validateAllowRestrict()
```

Comment:
It looks like validateAllowRestrict is only getting called for annotation overrides and not when the config is coming from viper.

Reply:
:facepalm: DOH

Reply:
Maybe we need to add a validation loop in the workflow functions? That's about the best suggestion I could think up.

Reply:
Oh, perhaps not.